scripts/build-xml

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+set -e
+
+# cli args
+in_dir='src/tiler/src/config/'
+out_dir='transpiled-configs/'
+s3_bucket=''
+node_env='production'
+
+function usage() {
+    echo -n "Usage: $(basename "${0}") -i <path> -o <path>
+Transpiles a directory's worth of MML+MSS files into Tilegarden-readable XML
+Options:
+    -h, --help          Display this help text
+    -i, --in <path>     Relative path to input directory [${in_dir}]
+    -o, --out <path>    Relative path to output directory [${out_dir}]
+    -b, --bucket <name> Uploads XML to the specified S3 bucket, instead of saving to disk (requires AWS CLI installed on your machine, uses credentials from .env)
+    -d, --development   Build using development variables, instead of production ones
+"
+    exit 0
+}
+
+function main() {
+    # if s3_bucket has been set, send all files to a temporary directory
+    # rather than the default out_dir
+    if [ ! -z ${s3_bucket} ]; then
+        out_dir=$(mktemp -d)
+    fi
+
+    # transpile files and save in the out_dir directory
+    for file in $(echo "${in_dir}/*")
+    do
+        ext="${file##*.}"
+        if [ "$ext" == "mml" ]; then
+            # get output path
+            filename="${file##*/}"
+            base="${filename%%.*}"
+            outPath="${out_dir}/${base}.xml"
+
+            mkdir -p "${out_dir}"
+
+            echo "Transpiling ${file} => ${outPath}"
+            docker-compose run -e "NODE_ENV=${node_env}" -v "$PWD/${in_dir}:/home/tiler/configs" tiler ./scripts/build-xml.sh "/home/tiler/configs/${base}.mml" "$(< ${file})" > ${outPath}
```

Comment:
Are there differences between this section and `build-all-xml.sh` besides the output directory?  Would it work to map that to a volume as well and make this command call `build-all-xml.sh` instead of repeating it?

Reply:
When docker creates files in a volume, it doesn't necessarily grant read/write access to the executor of the initial script, so I tried to circumvent this by echoing the output from `build-xml.sh` into the non-docker context.

Reply:
Oh right, the permissions.  Would be nice to not have to keep working around those.  But yeah, this makes sense.